### PR TITLE
xds:Allow big cluster total weight

### PR DESCRIPTION
**Pull request by larry-safran**
Allow a cluster's total weight to be in the range between max signed int and max unsigned int.

### CODE REVIEW

File: xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java [hunk 1]
```diff
@@ -34,21 +34,22 @@ final class WeightedRandomPicker extends SubchannelPicker {
...
 
   static final class WeightedChildPicker {
-    private final int weight;
+    private final long weight;
     private final SubchannelPicker childPicker;
 
-    WeightedChildPicker(int weight, SubchannelPicker childPicker) {
+    WeightedChildPicker(long weight, SubchannelPicker childPicker) {
       checkArgument(weight >= 0, "weight is negative");
+      checkArgument(weight <= 2L * Integer.MAX_VALUE, "weight is too large");
```
Comment (YifeiZhuang) [inline xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java:45]: max unsigned int is 2L * Integer.MAX_VALUE + 1. 
Use UnsignedInteger.MAX_VALUE.longValue()?

File: xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java [hunk 2]
```diff
@@ -93,12 +94,16 @@ public String toString() {
...
+    long totalWeight = 0;
     for (WeightedChildPicker weightedChildPicker : weightedChildPickers) {
-      int weight = weightedChildPicker.getWeight();
+      long weight = weightedChildPicker.getWeight();
+      checkArgument(weight >= 0, "weight is negative");
+      checkNotNull(weightedChildPicker.getPicker(), "childPicker is null");
       totalWeight += weight;
     }
     this.totalWeight = totalWeight;
+    checkArgument(totalWeight <= 2L * Integer.MAX_VALUE,
```
Comment (YifeiZhuang) [inline xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java:105]: ditto

File: xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java (hunk 1 above)
Comment (larry-safran) [reply xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java]: done.

File: xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java (hunk 2 above)
Comment (larry-safran) [reply xds/src/main/java/io/grpc/xds/WeightedRandomPicker.java]: Done.

**Review by YifeiZhuang — APPROVED** [review]

**Pull request merged**